### PR TITLE
Loki: Remove unused methods

### PR DESCRIPTION
**Pull request by ivanahuckova**
In this PR, I am:

- removing `processError` method as it wasn't used anywhere. Moreover, we have similar utility function  that we use in https://github.com/grafana/grafana/blob/main/public/app/plugins/datasource/loki/backendResultTransformer.ts#L104
- removing `getTime` method as it wasn't used anywhere. 
- I looked at the methods that are only used within the LokiDatasource class and marked them with `private`. That way we know which methods are now exported/used outside the class (although some of them are exported and only used in tests)
- In `LokiQueryEditor.tsx` we were using `datasource.getTimeRange();` to get timeRange, but data source query editors receive `range` as prop, so we can use that.
- And as I was working on `LokiQueryEditor` and `timeRange`, I fixed some spellcheck errors, mostly `timerange` => `timeRange`

### CODE REVIEW

File: public/app/plugins/datasource/loki/components/LokiQueryEditor.tsx [hunk 1]
```diff
@@ -29,17 +29,16 @@ export const testIds = {
 };
 
 export const LokiQueryEditor = React.memo<LokiQueryEditorProps>((props) => {
-  const { onChange, onRunQuery, onAddQuery, data, app, queries, datasource } = props;
+  const { onChange, onRunQuery, onAddQuery, data, app, queries, datasource, range } = props;
```
Comment (matyax) [inline public/app/plugins/datasource/loki/components/LokiQueryEditor.tsx:32]:
```suggestion
  const { onChange, onRunQuery, onAddQuery, data, app, queries, datasource, range: timeRange } = props;
```

And you can keep the same name it used to have, which I think is a better fit for it.

Comment (ivanahuckova) [reply public/app/plugins/datasource/loki/components/LokiQueryEditor.tsx]: Good idea! Fixed in https://github.com/grafana/grafana/pull/75106/commits/443764046c509ba1c399274525955d2e38ed6a8a

**Review by matyax — COMMENTED** [review]
👏 👏  great clean up!

**Comment by ivanahuckova** [comment]
@matyax I added 1 more commit as I realized that we have many types that we don't use anymore - https://github.com/grafana/grafana/pull/75106/commits/6334f4b909e0a261f0f9bfefa11da522e23fb12c. Would you give it another review?

File: public/app/plugins/datasource/loki/types.ts [hunk 2]
```diff
@@ -1,25 +1,9 @@
-import { DataQuery, DataQueryRequest, DataSourceJsonData, QueryResultMeta, ScopedVars, TimeRange } from '@grafana/data';
+import { DataQuery, DataQueryRequest, DataSourceJsonData, TimeRange } from '@grafana/data';
 
 import { Loki as LokiQueryFromSchema, LokiQueryType, SupportingQueryType, LokiQueryDirection } from './dataquery.gen';
 
 export { LokiQueryDirection, LokiQueryType, SupportingQueryType };
 
-export interface LokiInstantQueryRequest {
```
Comment (matyax) [inline public/app/plugins/datasource/loki/types.ts:7]: Wow, so many unused types.

**Review by matyax — APPROVED** [review]
🏆

File: public/app/plugins/datasource/azuremonitor/dashboards/postgresFlexibleServer.json [hunk 3]
```diff
@@ -1,8487 +1,7668 @@
...
-        "hideTimeOverride": true,
-        "id": 4,
-        "links": [],
-        "options": {
-          "colorMode": "background",
-          "graphMode": "none",
-          "justifyMode": "auto",
-          "orientation": "auto",
-          "reduceOptions": {
-            "calcs": [
```
Comment (ivanahuckova) [inline public/app/plugins/datasource/azuremonitor/dashboards/postgresFlexibleServer.json:180]: @aangelisc I see that this PR https://github.com/grafana/grafana/pull/74953 got merged yesterday. When I merged `main` the linter fixed bunch of extra spaces in this json file. I went thought all of the changes ( I recommend turning on `Hide whitespace` in this PR) an only whitespace/space/newlines got removed based on our linter.

Comment (aangelisc) [reply public/app/plugins/datasource/azuremonitor/dashboards/postgresFlexibleServer.json:180]: Hey @ivanahuckova, thanks for this 😊 it was an external PR so the linter config may have been invalid. Thank you for fixing!

**Pull request merged**